Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -1104,3 +1302,30 @@ string function SanitizeDisplayName( string displayName )
 	}
 	return result
 }
+
+// nothing in the game uses the format "Table.r/g/b/a"... wtf is the point of this function
+table function StringToColors( string colorString, string delimiter = " " )
```

Comment:
Not a fan of the untyped table stuff. It would probably be better to make a struct for this:
```
struct Color
{
    char r;
    char g;
    char b;
    char a;
}
```

Reply:
Ok, but have to use int instead. char is not supported in squirrel.
``` c
struct Color
{
    int r
    int g
    int b
    int a
}

Color function StringToColors( string colorString, string delimiter = " " )
{
	Color color
	array<string> tokens = split( colorString, " " )
	if ( tokens.len() < 3 || tokens.len() > 4)
	{
		throw "The length of tokens should be 3 or 4, but it is " + tokens.len()
	}

	color.r = int(clamp(int(tokens[0]), 0, 255))
	color.g = int(clamp(int(tokens[1]), 0, 255))
	color.b = int(clamp(int(tokens[2]), 0, 255))

	if ( tokens.len() == 4 )
		color.a = int(clamp(int(tokens[3]), 0, 255))
	else
		color.a = 255

	return color
}
```